File: app/src/main/java/com/prueba/mytodolist/database/DateConverter.java
package com.prueba.mytodolist.database;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

public class DateConverter {

    //We use converters, so we can store non primitive objects in our database.

    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/com/prueba/mytodolist/database/DateConverter.kt
package com.prueba.mytodolist.database

import androidx.room.TypeConverter
import java.text.SimpleDateFormat
import java.time.LocalDateTime
import java.util.*

//We use converters, so we can store non primitive objects in our database.

class DateConverter {

    companion object {

        private val DATE_FORMAT: String = "EEE, dd-MM-YYYY, HH:mm aaa"
        val dateFormat: SimpleDateFormat = SimpleDateFormat(DATE_FORMAT, Locale.getDefault())

        @TypeConverter
        @JvmStatic
        fun timestampToDate(timestamp: Long?): Date? {
            return when(timestamp) {
                null -> null
                else -> Date(timestamp)
            }
        }

        @TypeConverter
        @JvmStatic
        fun dateToTimestamp(date: Date?): Long? {
            return when (date){
                null -> null
                else -> date.time
            }
        }
    }
}